azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -87,17 +88,26 @@ def serialize(self) -> str:
         return json.dumps(result)
 
     @classmethod
-    def deserialize(cls, problem_as_json: str, name: str):
+    def deserialize(
+            cls, 
+            problem_as_json: str, 
+            name: Optional[str] = None
+        ):
         """Deserializes the problem from a
             JSON string serialized with Problem.serialize()
 
         :param problem_as_json:
             The string to be deserialized to a `Problem` instance
         :type problem_as_json: str
-        :param name: The name of the problem
-        :type name: str
+        :param name: 
+            The name of the problem is optional, since it will try 
+            to read the serialized name from the json payload.
+            If this parameter is not empty, it will use it as the
+            problem name ignoring the serialized value.
+        :type name: Optional[str]
         """
         result = json.loads(problem_as_json)
+        name = name if name else result.get("name")
```

Comment:
In line with the above comment, a name for the problem is not a part of the problem schema.